### PR TITLE
Add RSpec/ScatteredLet cop

### PR DESCRIPTION
**Pull request by Darhazer**
@bquorning I have promised you this one ;)

btw it seems related to ScatteredSetup - @backus what do you think about combining them?

### CODE REVIEW

File: lib/rubocop/cop/rspec/scattered_let.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      #   describe Foo do
+      #     subject { Foo }
+      #     before { prepare }
+      #     let(:foo) { 1 }
+      #     let(:bar) { 2 }
+      #     let!(:baz) { 3 }
+      #   end
+      #
+      class ScatteredLet < Cop
+        MSG = 'Group all let/let! blocks in the example group togetger.'.freeze
```
Comment (backus) [inline lib/rubocop/cop/rspec/scattered_let.rb]: together

**Comment by backus** [comment]
@Darhazer they are similar so I am glad you also used the word `Scattered` but this should be its own cop I think.

**Comment by Darhazer** [comment]
Fixed the typo

**Comment by backus** [comment]
🎈

**Pull request merged**